types/src/v0/impls/state.rs

```diff
@@ -306,6 +321,23 @@ pub fn validate_proposal(
         });
     }
 
+    // Validate timestamp increasing.
+    if proposal.timestamp() < parent_header.timestamp() {
+        return Err(ProposalValidationError::InvalidTimestampNonIncrementing {
+            proposal_timestamp: proposal.timestamp(),
+            parent_timestamp: parent_header.timestamp(),
+        });
+    }
+
+    // Validate timestamp hasn't drifted too much from system time.
+    let system_time: u64 = OffsetDateTime::now_utc().unix_timestamp() as u64;
+    if proposal.timestamp().abs_diff(system_time) > 2 {
```

Comment:
Maybe a comment that we think 2 seconds is enough for nodes that are reasonably configured and that we may want to make this configurable later?

Reply:
https://github.com/EspressoSystems/espresso-sequencer/pull/2089/commits/0f4ebc0f1310cdc4e524908f6567ae369e06fcf4